### ISSUE TITLE
[FR] Use analog pins to read TMC driver Vrefs

### ISSUE THREAD

**Issue by NSchnitzler** [MarlinFirmware/Marlin#7627]
It would be a nice new feature to connect the Vref pins of the TMC Stepper driver to the free analog pins and read out the Voltage or direct the calculated motor current over the Terminal. This would be much easier than holding a Multimeter and a screwdriver to adjust the right current.

**Comment by teemuatlut** [comment]
You can use [M43](http://marlinfw.org/docs/gcode/M043.html) to read a pin value.

You can look into TMC2130 or TMC2208 if you want easier current management through a command.

**Comment by NSchnitzler** [comment]
I Have 5 TMC2208, but couldn't find anything about them in the configuration_adv. I found some code for the TMC26X/TMC2130, but they have an other communication protocol.

**Comment by teemuatlut** [comment]
I've got a standing [PR for TMC2208](https://github.com/MarlinFirmware/Marlin/pull/6345) but it's almost 5 months old and I'd actually advice against using it. You can try my [master development branch](https://github.com/teemuatlut/Marlin/tree/master). That's where the most up-to-date TMC support is and I occasionally rebase it whenever I'm doing firmware work.

**Comment by NSchnitzler** [comment]
Ok i will give It a try. One more question, don't you need a single Serial wire for each TMC2208 stepper driver in order to config each single device?

**Comment by teemuatlut** [comment]
Ideally yes, one serial port for each driver. But you can also wire them in parallel if you don't care about read capabilities and this way the drivers will have the same configuration. However the write only mode is a bit untested because that's not my personal configuration. There's software serial capabilities if you can't use hw serial.

**Comment by NSchnitzler** [comment]
I had a quick look at the TMC2208 Arduino library and your master dev branch. For me, i would prefer a single software Serial interface for each driver, so you can tune chopper frequency, active current, hold current, micro stepping and so on individually. The Aux-2 connector would fit nicely for that idea (single extruder printer). In principle i have the capabilities to add such a feature, but the whole project structure is a little bit confusing for me, because all files are really large. I am not really sure, wether it's really necessary to read back the values, but it is useful to program each driver on it own (reading back is (for me) not a must have).

**Comment by teemuatlut** [comment]
And you can do that without modifications to the firmware. Set the correct pins and solder the wires. 
![img_20170418_181729](https://user-images.githubusercontent.com/7762686/30186160-06e52846-942d-11e7-9b04-1d9cfc1e1738.jpg)

**Comment by detoxity** [comment]
Hello, how i can cofigurate this firmware for TRAMS (tmc5130).
I use #define HAVE_TMC2130. and define pins. but its not work.

**Comment by teemuatlut** [comment]
You can't. Use my [TRAMS development branch](https://github.com/teemuatlut/Marlin/tree/trams-updated) of Marlin. You also need [a supporting library](https://github.com/teemuatlut/TMCStepper). The current status of TRAMS is largely untested and I need another printer to better start testing.

**Comment by detoxity** [comment]
In this branch when i click home x (in repetier host) my axis z come down. and printer does not respond to movement commands.

**Comment by teemuatlut** [comment]
It may be that the current online version isn't up to date what I have locally. I'll try to upload a more recent version along with my own configuration some time soon. But as said very much WIP.
I'd also ask that you make a new issue at my repo if you have further questions or probems.

**Comment by detoxity** [comment]
Thank you, I'm looking forward to your new version.

**Comment by NSchnitzler** [comment]
@teemuatlut thanks for the dev branch, i had a look at it and it should do what i need. I will give it a try at the weekend.

**Comment by ModMike** [comment]
How do I wire the TMC2208s in parallel mode?

**Comment by boelle** [comment]
@NSchnitzler you still need this or has the built in support for tmc2208 done the trick?

**Comment by boelle** [comment]
wrong label

**Comment by InsanityAutomation** [comment]
Id say the current state of UART and SPI comms to these drivers covers this use case and this can be closed. @NSchnitzler feel free to chime in if thats incorrect.

**Comment by NSchnitzler** [comment]
For me is everything working fine now.
